applications/PfemFluidDynamicsApplication/custom_utilities/calculate_wave_height_utility.cpp

```diff
@@ -46,18 +46,26 @@ CalculateWaveHeightUtility::CalculateWaveHeightUtility(
     mMeanWaterLevel = ThisParameters["mean_water_level"].GetDouble();
     double search_tolerance = ThisParameters["search_tolerance"].GetDouble();
     double relative_search_radius = ThisParameters["relative_search_radius"].GetDouble();
-    double mean_elem_size = block_for_each<SumReduction<double>>(
+    mMeanElementSize = block_for_each<SumReduction<double>>(
         mrModelPart.Elements(), [](Element& rElement) {
         return rElement.GetGeometry().Length();
     });
-    mean_elem_size /= mrModelPart.NumberOfElements();
-    mSearchRadius = relative_search_radius * mean_elem_size + search_tolerance;
+    mMeanElementSize /= mrModelPart.NumberOfElements();
+    mRelativeSearchRadius = relative_search_radius;
+    mSearchRadiusTolerance = search_tolerance;
 }
 
-double CalculateWaveHeightUtility::Calculate(const array_1d<double,3>& rCoordinates) const
+double CalculateWaveHeightUtility::Calculate(const array_1d<double,3>& rCoordinates, const double& rMeshSize) const
 {
     KRATOS_TRY
 
+    double search_radius = 0.0;
+    if (rMeshSize < std::numeric_limits<double>::epsilon()) {
+        search_radius = mRelativeSearchRadius * mMeanElementSize + mSearchRadiusTolerance;
+    } else {
+        search_radius = mRelativeSearchRadius * rMeshSize + mSearchRadiusTolerance;
```

Comment:
Instead of adding a new parameter, you can use the existing parameters. I.e., settings the `search_radius_tolerance` to the desired element size, will produce the same results.